packages/build-utils/package.json

```diff
@@ -19,7 +19,7 @@
     "vite": "^4.3.9"
   },
   "peerDependencies": {
-    "typescript": "^4.9.5",
+    "typescript": "~5.1.6",
```

Comment:
Tilde locks the major and minor versions, whereas caret only locks the major version. See the PR description for why this PR doesn't upgrade to TS 5.2.